### ISSUE TITLE
Mark `WebBrowser` and all related (e.g., HTML* types) as obsolete

### ISSUE THREAD

**Issue by RussKie** [dotnet/winforms#6964]
### Is your feature request related to a problem? Please describe

The `WebBrowser` control is based on IE Trident engine, which is almost universally unusable for general web browsing these days. It is also based on COM/ActiveX, which poses difficulties in trimming/native AOT scenarios.
Also, Internet [Explorer 11 has been retired](https://blogs.windows.com/windowsexperience/2022/06/15/internet-explorer-11-has-retired-and-is-officially-out-of-support-what-you-need-to-know/) and is officially out of support as of 15 June 2022. Whilst the underlying [Trident engine (mshtml.dll) isn't being removed](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/internet-explorer-11-desktop-app-retirement-faq/ba-p/2366549#toc-hId-894137149) (yet?) from Windows, the general guidance is to use the modern replacement, which is  [WebView2](https://docs.microsoft.com/microsoft-edge/webview2/) web browser control.

### Describe the solution you'd like and alternatives you've considered

For .NET 7 and .NET 8 decorate all web- and html-related types as obsolete (`error: false`) with a link to directing developers use  [WebView2](https://docs.microsoft.com/microsoft-edge/webview2/) control instead.

The obsoletion process entails the following:
- marking the API as obsolete, which will emit a build warning with a link to more information (refer to https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/better-obsoletion.md),
- documenting the obsoletion (the team will work with the docs team to make sure the docs are updated),
- enhancing the [.NET Upgrade Assistant](https://dotnet.microsoft.com/platform/upgrade-assistant) to guide developer to use the WebView2 API instead (the team will work with .NET Upgrade Assistant team).

The obsoletion will also happen in several stages:
- .NET 7/8 release - build warnings/documentation only
- (likely) subsequent release - build errors / implementation removal
- (possibly) subsequent future release - complete removal of API.

### Will this feature affect UI controls?

The affected controls will need to be made hidden from the designer.

**Comment by dreddy-work** [comment]
Team agrees principally.

**Comment by dotMorten** [comment]
Better idea: Replace just the internal implementation with the new browser control.

**Comment by RussKie** [comment]
I don't think this is feasible - the API surface is different, the types are different, the deployment model is different... It sounds similar to "I want to continue building VB6 apps, just plug the .NET runtime under the covers".

**Comment by kant2002** [comment]
Bye bye IE. let mark classes as obsolete. What are internal position? Does any discussion should happen for this plan?

**Comment by RussKie** [comment]
> Does any discussion should happen for this plan?

The team has discussed this proposal and reached an agreement that this proposal is worth pursuing further. However, the team doesn't currently have any cycles to work on this further. If you're keen, you're welcome to start working on this.

The obsoletion process entails the following:
- marking the API as obsolete, which will emit a build warning with a link to more information (refer to https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/better-obsoletion.md),
- documenting the obsoletion,
- enhancing the [.NET Upgrade Assistant](https://dotnet.microsoft.com/platform/upgrade-assistant) to help converting `MsgBox` to `MessageBox` API.

The obsoletion will also happen in several stages:
- current release - build warnings/documentation only
- (likely) next release - build errors / implementation removal
- future release - complete removal of API.

**Comment by kant2002** [comment]
> enhancing the [.NET Upgrade Assistant](https://dotnet.microsoft.com/platform/upgrade-assistant) to help converting MsgBox to MessageBox API.

Not clear what this is means. Does MsgBox uses HTML related classes internally? Where is .NET Upgrade Assistant lives. Ahh, I found [link](https://github.com/dotnet/upgrade-assistant) at the bottom of marketing page.

Can `MsgBox` => `MessageBox` API conversion treated as separate issue? Or at least work can be split in half that way. Also if you mark this as good first issue, maybe somebody except me can help.

**Comment by RussKie** [comment]
🤦‍♂️ Haha, I copied that from another issue and missed that bit. I now updated the text above.

**Comment by Svetoslav92** [comment]
Don't hurry up with that. Windows 8.1 is with extended support until January 10, 2023 and as we all know it it has IE11, Trident. Trident itself is fully working and functional in Windows 10,11 and not marked as obsolete. It makes sense to obsolete it here when Microsoft plans removing Trident ? I think this won't happen anytime soon as this will end HTA,Vbscript,Jscript and everything related. That means end of job of many system administrators and WW3

**Comment by paul1956** [comment]
WebView2 could be used as a replacement, but it needs a wrapper to add key back missing functionality. The biggest missing feature is the ability to view the document source which is trivial in WebBrowser.

**Comment by fffelix-jan** [comment]
I hope this ability can be added!

**Comment by hota99** [comment]
Webview2 uses many resources on a multi users environment (Such as UDF). We decided to keep good old webbrowser.

**Comment by paul1956** [comment]
@JeremyKuhne 
> Note that we will never remove API from the surface area. We also won't remove the ability to use this. When Windows removes Trident (likely never) then this won't work.

What does "Note that we will never remove API from the surface area. We also won't remove the ability to use this." mean in reality? 

1. The API will always exist but at some time in the future (if Windows removes Trident) it will not do anything.
2. Sometime in the future (just before Windows removes Trident) you will reimplement the broken parts and it will work transparently.
3. Something else.

If the second is true, why mark it as obsolete. If 3 that?

**Comment by JeremyKuhne** [comment]
@paul1956 we'll never remove surface area as that is a [binary breaking change](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-changes.md). Implementations may change or get removed, but I really doubt that will happen in the foreseeable future. I doubt we would ever reimplement this API over anything else as the cost would be quite high.

I don't make the decisions ultimately, but I know it would take an awful lot for Windows to remove Trident all together as it would be such a major impact. They still ship the VB6 runtime after all. :)

Ultimately, we'll probably do things to encourage you to move off of this, but we have no plans to make this functionality completely inaccessible.

**Comment by KlausLoeffelmann** [comment]
Closing this as out of scope.
